IdentityCore/src/webview/background/ios/MSIDBackgroundTaskManager.m

```diff
@@ -35,6 +35,32 @@ @interface MSIDBackgroundTaskManager()
 
 @end
 
+@interface MSIDBackgroundTaskData : NSObject
+
+@property (nonatomic) UIBackgroundTaskIdentifier backgroundTaskId;
+@property (nonnull, nonatomic) id callerReference;
+
+- (instancetype)init NS_UNAVAILABLE;
++ (instancetype)new NS_UNAVAILABLE;
+- (id)initWithTaskId:(UIBackgroundTaskIdentifier)backgroundTaskId caller:(id)caller;
+
+@end
+
+@implementation MSIDBackgroundTaskData
+
+-(id)initWithTaskId:(UIBackgroundTaskIdentifier)backgroundTaskId caller:(id)caller
+{
+    self = [super init];
+    if (self)
+    {
+        _backgroundTaskId = backgroundTaskId;
+        __typeof__(caller) __weak weakCallerReference = caller;
```

Comment:
I think a strong reference present in the cache might be dangerous. Although we can remove object from cache once stopOperation is called, currently we don't do so, to be extra careful I kept it as a weak ref instead of strong.

Reply:
Removed the need to pass in a reference.